secret-provider/deployment/deploy-secret-provider.sh

```diff
@@ -5,21 +5,8 @@
 
 : ${KUBE_NAMESPACE:=m4d-system}
 
-# DOCKER_PASSWORD is from the CI or from env variable
 deploy() {
     kubectl apply -n $KUBE_NAMESPACE -f secret-provider/secret-provider-rbac.yaml
-    if [[ -z $(kubectl get secrets cloud-registry  -n $KUBE_NAMESPACE --ignore-not-found) ]]; then
-        kubectl create secret docker-registry cloud-registry \
-                --docker-server="$DOCKER_HOSTNAME" \
-                --docker-username="$DOCKER_USERNAME" \
-                --docker-password="$DOCKER_PASSWORD" \
-                -n $KUBE_NAMESPACE
-    fi
-
-    kubectl patch serviceaccount secret-provider -p \
-    '{"imagePullSecrets": [{"name": "cloud-registry"}]}' \
-    -n $KUBE_NAMESPACE
-
```

Comment:
we should add an extra variable `DOCKER_SECRET` and if set we should create the k8s secret, thus supporting working against a private password protected registry

Reply:
This section was only required when the secret provider itself used a private image. Since it doesn't, I don't see the point.